src/thread.c

```diff
@@ -2217,6 +2217,28 @@ void ABTI_thread_print(ABTI_thread *p_thread, FILE *p_os, int indent)
     ABTU_free(prefix);
 }
 
+int ABTI_thread_stack_print(ABTI_thread *p_thread, FILE *p_os) {
+    void *p_stack = p_thread->attr.p_stack;
+    size_t i, stacksize = p_thread->attr.stacksize;
+    if (stacksize == 0 || p_stack == NULL) {
+        /* The thread is a main thread running on Pthreads, so we cannot
+         * dump the stack with p_stack. */
+        return ABT_ERR_THREAD;
+    }
+    for (i = 0; i < stacksize; i += 16) {
+        size_t j, jmax = (stacksize > i + 16) ? (i + 16) : stacksize;
+        for (j = i; j < jmax; j++) {
+            int val = (int)(((uint8_t *)p_stack)[j]);
```

Comment:
Still for manual unwinding, it would also be very helpful if the stack content is dumped based on the per-architecture address size (and endianness ?), thus it will be easy to detect call sequence return addresses.

Reply:
Thank you for your comments. I would appreciate if you would help its implementation. Since I don't have experience of manual stack unwinding, I don't know which format is helpful for developers in general.

Reply:
Instead of "per-architecture address size" I should have worded "per-architecture pointer size" !!!